src/core/Raycaster.js

```diff
@@ -46,11 +46,17 @@ function ascSort( a, b ) {
 
 function intersectObject( object, raycaster, intersects, recursive ) {
 
-	if ( object.layers.test( raycaster.layers ) ) {
+	var layerTest = object.layers.test( raycaster.layers );
+
+	if ( layerTest ) {
 
 		object.raycast( raycaster, intersects );
 
-		if ( recursive === true ) {
+	}
+
+	if ( recursive === true ) {
```

Comment:
I guess we can also get rid of the custom `recursive` in `Raycaster`?

Reply:
I'm not sure if I follow -- this `recursive` is the argument passed into the `intersectObject` function to indicate whether children should be traversed. It serves a different purpose. Are you suggesting getting rid of that argument?